src/observer.ts

```diff
@@ -342,32 +350,18 @@ export class Observer {
     arnsName: string;
     entropy: Buffer;
   }): Promise<ArnsNameAssessment> {
-    const referenceGatewayUrl = `https://${arnsName}.${this.referenceGatewayHost}/`;
-
-    const gotClient = client.extend({
-      headers: { 'X-AR-IO-Node-Release': this.nodeReleaseVersion },
-    });
+    if (this.referenceGatewayResolutionCache === undefined) {
```

Comment:
Can we instantiate the cache in the constructor to avoid the `undefined` check here?

Reply:
It’s possible if you’re ok setting `config.NUM_ARNS_NAMES_TO_OBSERVE_PER_GROUP * 2` to `cacheCapacity`.

Reply:
Yeah, I don't love the direct config reference either, but I think that's preferable to the `undefined` check here.

Reply:
We can't because we need to have access to the entropy Buffer, but we only generate it on `generateReport`.
Adding a TODO for now